service/service.go

```diff
@@ -108,19 +152,22 @@ func New(config Config) (*Service, error) {
 
 	newService := &Service{
 		// Dependencies.
-		Deployer: deployerService,
-		Version:  versionService,
+		Deployer:   deployerService,
+		HelmClient: helmClient,
+		Version:    versionService,
 	}
 
 	return newService, nil
 }
 
 type Service struct {
 	// Dependencies.
-	Deployer deployer.Deployer
-	Version  *version.Service
+	Deployer   deployer.Deployer
+	HelmClient helmclient.Interface
+	Version    *version.Service
 }
 
-func (s *Service) Boot() {
+func (s *Service) Boot(ctx context.Context) {
+	s.HelmClient.EnsureTillerInstalled(ctx)
```

Comment:
We need to handle error here. Let's make `Boot` returning error and handle it. IMO the service shouldn't start if it can't ensure tiller is up to date. It will be noop anyway.

Reply:
Good call. If it can't connect to tiller there is nothing useful it can do. I'll do the same for chart-operator.

Reply:
Witch chart-operator we can retry on every reconciliation loop.
But I would be fine to align approaches here. There is no reason to call `EnsureTillerInstalled` on every loop.

Reply:
I was being dumb chart-operator is already covered. See https://github.com/giantswarm/giantswarm/issues/4938#issue-392214229